allennlp/models/semantic_parsing/wikitables/wikitables_erm_semantic_parser.py

```diff
@@ -285,25 +285,34 @@ def forward(self,  # type: ignore
             outputs['similarity_scores'] = similarity_scores
             outputs['logical_form'] = []
             best_action_sequences = outputs['best_action_sequences']
+            outputs["best_action_sequence"] = []
+            outputs['debug_info'] = []
             agenda_indices = [actions_[:, 0].cpu().data for actions_ in agenda]
             for i in range(batch_size):
                 in_agenda_ratio = 0.0
                 # Decoding may not have terminated with any completed logical forms, if `num_steps`
                 # isn't long enough (or if the model is not trained enough and gets into an
                 # infinite action loop).
+                outputs['logical_form'].append([])
                 if i in best_action_sequences:
                     # Taking only the top action sequence.
-                    best_action_sequence = best_action_sequences[i][0]
-                    action_strings = [action_mapping[(i, action_index)] for action_index in best_action_sequence]
-                    try:
-                        self._has_logical_form(1.0)
-                        logical_form = world[i].get_logical_form(action_strings, add_var_function=False)
-                    except ParsingError:
-                        self._has_logical_form(0.0)
-                        logical_form = 'Error producing logical form'
-                    if example_lisp_string:
-                        self._denotation_accuracy(logical_form, example_lisp_string[i])
-                    outputs['logical_form'].append(logical_form)
+                    got_top_action_sequence = False
+                    for action_sequence in best_action_sequences[i]:
```

Comment:
Can't you just do

```python
for j, action_action in enumerate(best_action_sequences[i]):
    if j == 0:
        # block for top action sequence here
    # block for what you do every time here
```

Then you don't need this `got_top_action_sequence` flag.

Reply:
That's correct. Changed this.